### ISSUE TITLE
[Security] Node16 is EOL

### ISSUE THREAD

**Issue by billyjbryant** [nightscout/cgm-remote-monitor#8207]
**Describe the bug**
NodeJS 16 is EOL as of September 2023 and is no longer receiving security updates. Efforts should be made to modernize the application to upgrade to NodeJS 20, as NodeJS 18 will be EOL in April 2025 where as NodeJS 20 has support through April 2026

**To Reproduce**
N/A

**Expected behavior**
N/A

**Screenshots**
N/A

**Your setup information**
N/A

**Additional context**
* https://endoflife.date/nodejs

**Comment by psonnera** [comment]
IMHO EOL doesn't mean it won't work anymore. There is constant ongoing work to maintain Nightscout and you're welcome to join the developers. Some obsolete components are being dropped to allow moving forward with node.js.

**Comment by billyjbryant** [comment]
EOL doesn't mean it doesn't work, you are correct. What it does mean is that it is no longer considered secure as there will be no more security updates for EOL versions. 

Security when you are dealing with medical devices even if just for monitoring is paramount.

**Comment by ninelore** [comment]
Did someone pick up on migrating yet? If not i will have a look

**Comment by psonnera** [comment]
If you want to join the dev effort, I don't believe anybody would complain.

**Comment by psonnera** [comment]
I am not dev: only doc.
I think it's in the roadmap. share is already moved to nightscout connect AFAIK.
minimed is killed by 2FA for now

**Comment by bewest** [comment]
Yes, significant effort is underway here. Nightscout-connect can replace
both of those and is designed to help constrain dependencies.  There is
also a patch in dev to augment the supported versions to whatever is in LTS.
The bigger issue is migrating momentjs, d3, and benv.  Additional help
migrating these would be amazing.  I started a branch evaluating playwright
for testing.

Thanks,
-bewest

**Comment by ninelore** [comment]
> The bigger issue is migrating momentjs, d3, and benv. Additional help migrating these would be amazing. I started a branch evaluating playwright for testing.

I'd suggest a tracking issue for this if it doesnt already exists

**Comment by ninelore** [comment]
> To expand a bit on this issue: EOL Node is currently the biggest hurdle to overcome when deploying on FreeBSD. I'd be willing as well to help out with the migration.

iirc new node seems to work, but seems to be neither officially tested not endorsed atm

latest discussion here was more about old and vulnerable dependencies
